src/AaveV2_Eth_TreasuryManagement_20230308/COWSwapper20230801.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.19;
+
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {SafeERC20} from 'solidity-utils/contracts/oz-common/SafeERC20.sol';
+import {AaveV2Ethereum, AaveV2EthereumAssets} from 'aave-address-book/AaveV2Ethereum.sol';
+import {AaveGovernanceV2} from 'aave-address-book/AaveGovernanceV2.sol';
+import {IMilkman} from './interfaces/IMilkman.sol';
+
+contract COWSwapper {
+  using SafeERC20 for IERC20;
+
+  error InvalidCaller();
+
+  address public constant ALLOWED_CALLER = 0xA519a7cE7B24333055781133B13532AEabfAC81b;
+  address public constant MILKMAN = 0x11C76AD590ABDFFCD980afEC9ad951B160F02797;
+  address public constant CHAINLINK_PRICE_CHECKER = 0xe80a1C615F75AFF7Ed8F08c9F21f9d00982D666c;
+
+  function swap(address fromToken, address oracleFrom, uint256 slippage) external {
+    if (msg.sender != AaveGovernanceV2.SHORT_EXECUTOR) revert InvalidCaller();
+
+    uint256 balance = IERC20(fromToken).balanceOf(address(this));
+
+    IERC20(fromToken).safeApprove(MILKMAN, balance);
```

Comment:
Better to avoid using `safeApprove()` as it has been deprecated.

Reply:
@brotherlymite how do you approve USDT without `safeApprove`? with `safeIncreaseAllowance`?
USDT fails with regular `approve`.

Reply:
@efecarranza You can check `forceApprove()` https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/utils/SafeERC20.sol#L76.

Reply:
you can import from main, as we have updated the libs so in solidity-utils has the new version of SafeERC20 with the forceApprove method

Reply:
thank you!